7_factory_wrap/node_4.x/addon.cc

```diff
@@ -0,0 +1,17 @@
+#include <node.h>
+#include "myobject.h"
+
+using namespace v8;
+
+void CreateObject(const FunctionCallbackInfo<Value>& args) {
+  Isolate* isolate = args->GetIsolate();
+  MyObject::NewInstance(args);
+}
+
+void InitAll(Handle<Object> exports, Handle<Object> module) {
+  MyObject::Init();
```

Comment:
I would change this so that you pass `exports->GetIsolate()` to `MyObject::Init()`.